Makefile

```diff
@@ -44,15 +32,30 @@ install-bin:
 		echo "Otherwise, pac4cli may fail to work properly."; \
 	fi
 
-	install -D -m 755 pac4cli.sh $(DESTDIR)etc/profile.d/pac4cli.sh
+install-self-contained: check-prev-proxies
+	virtualenv -p $(PYTHON) --system-site-packages $(DESTDIR)/opt/pac4cli
+	$(DESTDIR)/opt/pac4cli/bin/pip install -r requirements.txt
+	PYTHON=$(DESTDIR)/opt/pac4cli/bin/python make -C pacparser/src install-pymod
+
+	install -m 755 -d $(DESTDIR)/opt/pac4cli
+	install -m 644 main.py proxy.py $(DESTDIR)/opt/pac4cli
+	install -m 755 uninstall.sh $(DESTDIR)/opt/pac4cli
+	install -D -m 644 pac4cli.service $(DESTDIR)/lib/systemd/system/pac4cli.service
+	install -D -m 755 trigger-pac4cli $(DESTDIR)/etc/NetworkManager/dispatcher.d/trigger-pac4cli
+	install -D -m 755 pac4cli.sh $(DESTDIR)/etc/profile.d/pac4cli.sh
 
-install: install-env install-bin
+install: check-prev-proxies
+	install -D -m 755 main.py $(DESTDIR)/$(PREFIX)/bin/pac4cli
+	install -D -m 644 proxy.py $(DESTDIR)/$(PREFIX)/lib/python3.6/site-packages
+	install -D -m 644 pac4cli.service $(DESTDIR)/$(PREFIX)/lib/systemd/system/pac4cli.service
+	install -D -m 755 trigger-pac4cli $(DESTDIR)/$(PREFIX)/etc/NetworkManager/dispatcher.d/trigger-pac4cli
+	install -D -m 755 pac4cli.sh $(DESTDIR)/$(PREFIX)/etc/profile.d/pac4cli-proxy.sh
 
-install-debian: install-bin
+install-debian: install
```

Comment:
the `check-pre-proxies` is run by `install` target only for the case when a user (or us) install it on the host directly (running `make install`) from the source dir.

But since we will likely have a configure script (for specializing the `.service` template) we can move this check to that script.

As for Arch, it is doing that already (`PYTHON=python3 make DESTDIR="${pkgdir}" install`)